bentoml/_internal/runner/runner.py

```diff
@@ -99,7 +100,13 @@ def __init__(
             method_configs: per method configs
         """
 
-        name = runnable_class.__name__ if name is None else name
+        if name is None:
+            name = runnable_class.__name__
+        elif not validate_tag_str(name):
```

Comment:
```suggestion
        if name is None:
            name = runnable_class.__name__
        
        if not validate_tag_str(name):
```

Reply:
The runner name is used as a Kubernetes labels value, and here are the requirements: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set validate_tag_str is perfect for this check, but a Python class name may not meet the requirment